### DISCUSSION TITLE
gentoo compile

### DISCUSSION THREAD

**Discussion by drforbin** [brndnmtthws/conky#2102] [Q&A]
I am running version 17.1 and it compiles fine on gentoo.
I cannot get 1.21 to compile. 

Please help

**Comment by Caellian** [comment]
Please provide compilation logs.

**Comment by drforbin** [comment]
I Will thank you....I'll get to it tonight

**Comment by drforbin** [comment]
Here is compile log.
[build.log](https://github.com/user-attachments/files/18054222/build.log)

**Reply by drforbin** [reply]
Anything?

**Reply by Caellian** [reply]
tolua++ fails to build because the linker can't find lua.

tolua++ hasn't been touched in a very long time, so it's something to do with your environment.

**Comment by drforbin** [comment]
I have lua installed

**Reply by Caellian** [reply]
Which version of Lua do you have installed? Linked version is the one provided by `/usr/local/lib/liblua.so` so check with your package manager whether that's the Lua library you expect to link and let me know the version.

The linker says linked Lua library is missing:
- `lua_setiuservalue`/`lua_getiuservalue` - added in Lua 5.4
  - `toluapp` in conky 3rdparty uses `lua_setuservalue`/`lua_getuservalue` versions which are available in 5.3
  - this lua version is asserted at configure phase with CMake, so a linking error is very odd unless you're using a patched version of conky
- `lua_rotate` - added in Lua 5.3

You're trying to compile an older version of conky (1.21.7), do you mind cloning and trying to build the latest version with no modifications? Main should be fine, but try [1.21.9](https://github.com/brndnmtthws/conky/tree/v1.21.9) if you'd prefer a release.
